module/extension/FileWatcher/src/FileWatcher.cpp

```diff
@@ -234,7 +234,7 @@ namespace module::extension {
                 for (const auto& element : utility::file::listDir(path)) {
                     // Set our thread local event details
                     FileWatch watch;
-                    watch.path   = path + "/" + element;
+                    watch.path   = path.append("/").append(element);
```

Comment:
This is once again another bug as you're modifying the path element on each loop, use `std::filesystem`